traversal/executor/pattern_executor.rs

```diff
@@ -399,16 +439,16 @@ impl SortedExecutor {
         loop {
```

Comment:
Am I missing something or can this be simplified to?

```
let mut failed = false; // Don't need retry.
while !failed {
 ...
}
if failed { ... }
else { /* !retry is implicitly true*/ }
```

Reply:
I have the feeling the whole loop could be simplified/optimised with a max-heap.

Reply:
It may be, we could look at it monday